### PR TITLE
[FLOC-1607] Encode subjectAltName correctly when given hostname is actually an IP

### PR DESCRIPTION
**Pull request by itamarst**
And add dependency on `service_identity`.

### CODE REVIEW

File: flocker/ca/test/test_ca.py [hunk 1]
```diff
@@ -352,14 +352,24 @@ def test_certificate_subject_control_service(self):
...
         The generated certificate has a subjectAltName containing the given
-        hostname.
+        hostname as a DNS record.
         """
         assert_has_extension(self, self.credential.credential,
                              b"subjectAltName", b"DNS:control.example.com")
 
+    def test_subjectAltName_ipv4(self):
+        """
+        The generated certificate has a subjectAltName containing the given
```
Comment (adamtheturtle) [inline flocker/ca/test/test_ca.py]: Reading the test docstring for this and `test_subjectAltName_ipv4`, they can't both be true. Put "if ..." at the end of each.

File: setup.py [hunk 2]
```diff
@@ -90,6 +90,10 @@
     "pytz",
     "characteristic >= 14.1.0",
     "Twisted == 15.1.0",
+    # TLS support libraries for Twisted:
+    "service_identity == 14.0.0",
```
Comment (adamtheturtle) [inline setup.py]: Is it more important with these dependencies than others that they are kept up to date (just because they relate to security)?

**Comment by adamtheturtle** [comment]
Thanks @itamarst, I'm accepting this with the assumption that with just the additional dependencies and without the additional code, Buildbot fails (like my local machine failed yesterday).

File: setup.py (hunk 2 above)
Comment (itamarst) [inline setup.py]: In general they ought to be kept up to date, yes, although we don't actually rely on service_identity or idna for security.

**Comment by itamarst** [comment]
It does fail, yeah. I also filed an issue for adding a test for the underlying use case that this would break (3rd party HTTPS clients).

**Pull request merged**